Singularity Support in Makeflow

Nekel-Seyew opened the pull request:
Added the ability to wrap tasks in singularity.

dthain commented on makeflow/src/makeflow.c:
```diff
@@ -1757,10 +1772,14 @@ int main(int argc, char *argv[])
-		system(cmd);
-		free(cmd);
-	}
+            char *cmd = string_format("rm %s", CONTAINER_SH);
```
Simplify to unlink(CONTAINER_SH)

dthain replied:
For consistency: CONTAINER_SH -> CONTAINER_DOCKER_SH

dthain commented on makeflow/src/dag.h:
```diff
@@ -32,6 +32,7 @@ struct dag {
 	struct set *export_vars;           /* List of variables with prefix export. (these are setenv'ed eventually). */
 	struct set *special_vars;          /* List of special variables, such as category, cores, memory, etc. */
 	category_mode_t allocation_mode;   /* One of CATEGORY_ALLOCATION_MODE_{FIXED,MAX_THROUGHTPUT,MIN_WASTE} */
+        int wrapped_in_container;          /* Indicates whether or not the current makeflow is wrapped in a container*/
```
As far as I can tell, this doesn't get used.
And, it's an alias for CONTAINER_MODE_NONE.
Can it be removed?

dthain commented:
Apart from a few nitpicks, this is looking good.  Please address those, squash to one commit, and solve conflicts.

Nekel-Seyew commented:
Note: this code requires the custom version of Singularity I modified and installed on the disc servers. I will update this code when singularity 2.2 that includes my changes is release. The difference is that I use the --change-home </path/to/home> switch I added, and the v2.2 uses -H </path/to/home>. V2.2 is in testing, soon to be released as of this writing (sept 30)

dthain commented:
@Nekel-Seyew is this ready to merge?

dthain commented:
I nearly forgot, please update `doc/makeflow.html` and `doc/man/makeflow.m4` appropriately for the new features.
